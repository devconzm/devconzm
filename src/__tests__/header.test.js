import React from "react";
import { render } from "react-testing-library";
import Header from "../components/Header";

describe("Header Component", () => {
  const { getByTestId} = render(<Header />);
  const links = getByTestId("head-list");
  it("should render a list of links", () => {
    expect(links.children.length).toBe(7);
  });
  it("should contain the title DevCon19", () => {
    expect(links).toHaveTextContent("DevCon19");
  });
  it('should match the snapshot', () => {
      expect(render(<Header />)).toMatchSnapshot()
  });
});

