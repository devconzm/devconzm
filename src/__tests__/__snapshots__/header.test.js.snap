// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        id="header"
      >
        <nav
          class="navbar fixed  navbar-expand-lg navbar-light top-navbar"
          data-toggle="sticky-onscroll"
        >
          <div
            class="container"
          >
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-target="#navbarSupportedContent"
              data-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="row"
            >
              <div
                class="col"
                id="banner-img"
              >
                <a
                  class="navbar-brand"
                  href="#header"
                >
                  <img
                    alt="DevConZM19"
                    class="img-fluid"
                    src="dev1.png"
                    title="DevConZM19"
                  />
                </a>
              </div>
              <div
                class="col"
              >
                <div
                  class="pr-4 pt-2 col-12 ml-auto"
                >
                  <div
                    class="collapse navbar-collapse justify-content-end"
                    id="navbarSupportedContent"
                  >
                    <ul
                      class="navbar-nav pull-right smooth-scroll"
                      data-testid="head-list"
                    >
                      <li
                        class="nav-item"
                      >
                        <a
                          class="nav-link active"
                          href="#venue"
                        >
                          Venue
                        </a>
                      </li>
                      <li
                        class="nav-item"
                      >
                        <a
                          class="nav-link"
                          href="#partners"
                        >
                          Partners
                        </a>
                      </li>
                      <li
                        class="nav-item"
                      >
                        <a
                          class="nav-link"
                          href="#speakers"
                        >
                          Speakers
                        </a>
                      </li>
                      <li
                        class="nav-item"
                      >
                        <a
                          class="nav-link"
                          href="#tickets"
                        >
                          Tickets
                        </a>
                      </li>
                      <li
                        class="nav-item"
                      >
                        <a
                          class="nav-link"
                          href="#about"
                        >
                          About
                        </a>
                      </li>
                      <li
                        class="nav-item"
                      >
                        <a
                          class="btn btn-outline-primary devcon"
                          href="#devcon"
                        >
                          DevCon19
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
                <div
                  class="pl-4 pt-5 col-12"
                >
                  <img
                    alt="DevConZM19"
                    class="img-fluid"
                    src="dev2.png"
                    title="DevConZM19"
                  />
                </div>
                <div
                  class="pl-4 pt-5 event-date col-12"
                >
                  17th
                  <br />
                  November
                  <br />
                  2019
                  <br />
                </div>
                <div
                  class="pl-5 pt-5 col-12 text-center"
                >
                  <button
                    class="btn btn-outline-primary devcon"
                    type="button"
                  >
                    Register Now
                  </button>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      id="header"
    >
      <nav
        class="navbar fixed  navbar-expand-lg navbar-light top-navbar"
        data-toggle="sticky-onscroll"
      >
        <div
          class="container"
        >
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="row"
          >
            <div
              class="col"
              id="banner-img"
            >
              <a
                class="navbar-brand"
                href="#header"
              >
                <img
                  alt="DevConZM19"
                  class="img-fluid"
                  src="dev1.png"
                  title="DevConZM19"
                />
              </a>
            </div>
            <div
              class="col"
            >
              <div
                class="pr-4 pt-2 col-12 ml-auto"
              >
                <div
                  class="collapse navbar-collapse justify-content-end"
                  id="navbarSupportedContent"
                >
                  <ul
                    class="navbar-nav pull-right smooth-scroll"
                    data-testid="head-list"
                  >
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link active"
                        href="#venue"
                      >
                        Venue
                      </a>
                    </li>
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="#partners"
                      >
                        Partners
                      </a>
                    </li>
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="#speakers"
                      >
                        Speakers
                      </a>
                    </li>
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="#tickets"
                      >
                        Tickets
                      </a>
                    </li>
                    <li
                      class="nav-item"
                    >
                      <a
                        class="nav-link"
                        href="#about"
                      >
                        About
                      </a>
                    </li>
                    <li
                      class="nav-item"
                    >
                      <a
                        class="btn btn-outline-primary devcon"
                        href="#devcon"
                      >
                        DevCon19
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
              <div
                class="pl-4 pt-5 col-12"
              >
                <img
                  alt="DevConZM19"
                  class="img-fluid"
                  src="dev2.png"
                  title="DevConZM19"
                />
              </div>
              <div
                class="pl-4 pt-5 event-date col-12"
              >
                17th
                <br />
                November
                <br />
                2019
                <br />
              </div>
              <div
                class="pl-5 pt-5 col-12 text-center"
              >
                <button
                  class="btn btn-outline-primary devcon"
                  type="button"
                >
                  Register Now
                </button>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
